[Standalone Mode] ConfigProvider should able to work when secure vault is not configured

**At the moment carbon-config has a tight dependency with carbon-secvault. It is Ok to have a tight dependency with secvault but there are use cases where we don't require secure vault but only required config provider functionality. But, it seems config provide will not work unless we provide necessary files (master-keys.yaml and secret.properties) which are related secvault to the processing. We need to fix this..** 

Below is the error that we could observe. 

[main] ERROR org.wso2.carbon.databridge.agent.AgentHolder  - Unable to complete initialization of agents.Error while loading the configuration file /home/mohan/wso2/source-code/mohan-repo/carbon-analytics-common/components/data-bridge/org.wso2.carbon.databridge.agent/src/test/resources/data.agent.config.yaml
org.wso2.carbon.databridge.agent.exception.DataEndpointAgentConfigurationException: Error while loading the configuration file /home/mohan/wso2/source-code/mohan-repo/carbon-analytics-common/components/data-bridge/org.wso2.carbon.databridge.agent/src/test/resources/data.agent.config.yaml
	at org.wso2.carbon.databridge.agent.AgentHolder.loadConfiguration(AgentHolder.java:182)
	at org.wso2.carbon.databridge.agent.AgentHolder.<init>(AgentHolder.java:65)
	at org.wso2.carbon.databridge.agent.AgentHolder.getInstance(AgentHolder.java:81)
	at org.wso2.carbon.databridge.agent.DataPublisher.<init>(DataPublisher.java:125)
	at org.wso2.carbon.databridge.agent.test.binary.OneEndPointDPBinaryTest.testInvalidAuthenticationURLs(OneEndPointDPBinaryTest.java:189)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:659)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:281)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:121)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
Caused by: org.wso2.carbon.databridge.agent.exception.DataEndpointAgentConfigurationException: Error in when loading databridge agent configuration
	at org.wso2.carbon.databridge.agent.AgentHolder.loadConfiguration(AgentHolder.java:150)
	... 31 more
Caused by: org.wso2.carbon.config.ConfigurationException: Error while loading securevault service
	at org.wso2.carbon.config.ConfigProviderFactory.getSecureVault(ConfigProviderFactory.java:106)
	at org.wso2.carbon.config.ConfigProviderFactory.getConfigProvider(ConfigProviderFactory.java:49)
	at org.wso2.carbon.databridge.agent.AgentHolder.loadConfiguration(AgentHolder.java:139)
	... 31 more
Caused by: org.wso2.carbon.secvault.exception.SecureVaultException: Error initializing securevault, secure configuration does not exist
	at org.wso2.carbon.secvault.SecureVaultUtils.getSecureVaultConfiguration(SecureVaultUtils.java:245)
	at org.wso2.carbon.secvault.SecureVaultUtils.resolveFileToString(SecureVaultUtils.java:224)
	at org.wso2.carbon.secvault.SecureVaultUtils.getSecureVaultConfig(SecureVaultUtils.java:88)
	at org.wso2.carbon.secvault.SecureVaultFactory.getSecureVault(SecureVaultFactory.java:49)
	at org.wso2.carbon.config.ConfigProviderFactory.getSecureVault(ConfigProviderFactory.java:103)
	... 33 more

--- a/components/org.wso2.carbon.config/src/main/java/org/wso2/carbon/config/ConfigProviderFactory.java
+++ b/components/org.wso2.carbon.config/src/main/java/org/wso2/carbon/config/ConfigProviderFactory.java
@@ -67,8 +67,7 @@ public class ConfigProviderFactory {
         }
         //check whether securevault is null. proceed if not null.
         if (secureVault == null) {
-            throw new ConfigurationException("No securevault service found. configuration provider will not be " +
-                    "initialized!");
+            logger.debug("No securevault service found. configuration provider will not be initialized!");
         }
         if (logger.isDebugEnabled()) {
             logger.debug("initialize config provider instance from configuration file: " + filePath.toString());

--- a/components/org.wso2.carbon.config/src/test/java/org/wso2/carbon/config/configprovider/ConfigProviderFactoryTest.java
+++ b/components/org.wso2.carbon.config/src/test/java/org/wso2/carbon/config/configprovider/ConfigProviderFactoryTest.java
@@ -81,11 +81,11 @@ public class ConfigProviderFactoryTest {
                 secureVault);
     }
 
-    @Test(description = "test case when securevault is not provided, when getting config provider", expectedExceptions
-            = ConfigurationException.class, expectedExceptionsMessageRegExp = "No securevault service found. " +
-            "configuration provider will not be initialized!")
+    @Test(description = "test case when securevault is not provided, when getting config provider")
     public void secureVaultNotProvidedTestCase() throws ConfigurationException {
-        ConfigProviderFactory.getConfigProvider(TestUtils.getResourcePath("conf", "Example.yaml").get(), null);
+        Assert.assertNotNull(
+                ConfigProviderFactory.getConfigProvider(TestUtils.getResourcePath("conf", "Example.yaml").get(), null),
+                "Couldn't initialize ConfigProvider without Secvault");
     }
 
     @Test(description = "test case for xml configuration file", expectedExceptions = ConfigurationException.class)

